cg/meta/orders/api.py

```diff
@@ -11,47 +11,148 @@
 
 from cg.apps.lims import LimsAPI
 from cg.apps.osticket import OsTicket
-from cg.meta.orders.balsamic_qc_submitter import BalsamicQCSubmitter
-from cg.meta.orders.balsamic_submitter import BalsamicSubmitter
-from cg.meta.orders.balsamic_umi_submitter import BalsamicUmiSubmitter
-from cg.meta.orders.fastq_submitter import FastqSubmitter
-from cg.meta.orders.fluffy_submitter import FluffySubmitter
-from cg.meta.orders.metagenome_submitter import MetagenomeSubmitter
-from cg.meta.orders.microsalt_submitter import MicrosaltSubmitter
-from cg.meta.orders.mip_dna_submitter import MipDnaSubmitter
-from cg.meta.orders.mip_rna_submitter import MipRnaSubmitter
-from cg.meta.orders.rml_submitter import RmlSubmitter
-from cg.meta.orders.rnafusion_submitter import RnafusionSubmitter
-from cg.meta.orders.sars_cov_2_submitter import SarsCov2Submitter
-from cg.meta.orders.submitter import Submitter
 from cg.meta.orders.ticket_handler import TicketHandler
-from cg.meta.orders.tomte_submitter import TomteSubmitter
 from cg.models.orders.order import OrderIn, OrderType
+from cg.services.orders.order_lims_service.order_lims_service import OrderLimsService
+from cg.services.orders.store_order_services.store_fastq_order_service import StoreFastqOrderService
+from cg.services.orders.store_order_services.store_generic_order import StoreGenericOrderService
+from cg.services.orders.store_order_services.store_metagenome_order import (
+    StoreMetagenomeOrderService,
+)
+from cg.services.orders.store_order_services.store_microbial_order import StoreMicrobialOrderService
+from cg.services.orders.store_order_services.store_pool_order import StorePoolOrderService
+from cg.services.orders.submitters.fastq_order_submitter import FastqOrderSubmitter
+from cg.services.orders.submitters.generic_order_submitter import GenericOrderSubmitter
+from cg.services.orders.submitters.metagenome_order_submitter import MetagenomeOrderSubmitter
+from cg.services.orders.submitters.microbial_order_submitter import MicrobialOrderSubmitter
+from cg.services.orders.submitters.order_submitter import (
+    OrderSubmitter,
+    ValidateOrderService,
+    StoreOrderService,
+)
+from cg.services.orders.submitters.pool_order_submitter import PoolOrderSubmitter
+from cg.services.orders.validate_order_services.validate_fastq_order import (
+    ValidateFastqOrderService,
+)
+from cg.services.orders.validate_order_services.validate_generic_order import (
+    ValidateGenericOrderService,
+)
+from cg.services.orders.validate_order_services.validate_metagenome_order import (
+    ValidateMetagenomeOrderService,
+)
+from cg.services.orders.validate_order_services.validate_microbial_order import (
+    ValidateMicrobialOrderService,
+)
+from cg.services.orders.validate_order_services.validate_pool_order import ValidatePoolOrderService
 from cg.store.store import Store
 
 LOG = logging.getLogger(__name__)
 
+order_service_mapping = {
+    OrderType.BALSAMIC: (
+        OrderLimsService,
```

Comment:
It seems like the LimsService is the same for all of them, why not instantiate it in the same way for all orders?

Reply:
yes, I am not sure about this at the moment. Something we should discuss. I am also unsure about using a factory instead of a registry.